### PR TITLE
feature: Add NSG and NSG rules.  BREAKING

### PR DESCRIPTION
**Pull request by Blankf**
With this release, you can add additional NSG's if required.
with this release, you can also add NSG rules from your plan towards those nsg rules.
Bastion subnet is added as well, it will generated the right NSG rules if you want an NSG on the subnet.

### CODE REVIEW

**Review by Dennisvandermeulen — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -54,52 +54,48 @@ This object describes the public IP configuration when creating Nat Gateway's wi
...
+    address_prefixes                = optional(list(string))
+    default_outbound_access_enabled = optional(bool, false)
+    delegate_to                     = optional(string, null)
     nat_gateway = optional(object({
       id = string
     }))
-    network_security_group = optional(object({
-      id = string
-    }))
+    name                          = optional(string)
```
Comment (mpwaldhorst) [inline variables.tf:66]: Nitpick: I see this in the other variables as well, wondering if we should just move the name always up to the top

File: variables.tf [hunk 2]
```diff
@@ -123,15 +124,37 @@ This object describes the subnets to create within the virtual network.
...
+      azure_default = true
     }
   }
-  ```hcl
+  "OtherSubnet" = {
+    address_prefixes                = ["100.0.4.0/24"]
+    default_outbound_access_enabled = false
+    no_nsg_association              = true
+  }
+  "AzureBastionSubnet" = {
```
Comment (mpwaldhorst) [inline variables.tf:152]: I'm wondering if some of the stuff in this module related to the Bastion should not be made part of the bastion mcaf module instead.

Any particular reason why we don't create the necessary subnet there? I guess that might be related to your address_prefixes?

Comment (Blankf) [reply variables.tf:152]: imo, you should create the networks from your networking module.
if you decide to not use the networking module, that up-to-you ofc.
but then you indeed need to split all your network creations from other locations.
but you should be able to deploy this from the networking module, thats its purpose.

Comment (Blankf) [reply variables.tf:152]: haven't checked that module, but yes we could make a not that you could use this module to create it.
but ofc it should not be required, people are still allowed to generate subnets elsewehere right?

Comment (mpwaldhorst) [reply variables.tf:152]: yeah just so people know that they need to do more than just use the bastion module in order for them to deploy a bastion

**Review by mpwaldhorst — APPROVED** [review]

**Pull request merged**